src/main/java/org/example/Main.java

```diff
@@ -122,338 +163,205 @@ public boolean isHappy() {
-        return users.contains(param) && check.check() || users.contains("-" + param) && !check.check();
+    private static boolean check(Collection<String> users, String param, Checkable check, MyNumber num) {
+        return users.contains(param) && check.check(num) || users.contains("-" + param) && !check.check(num);
     }
 
-    private enum Params {
-        EVEN(num -> !num.even),
-*//*        ODD,
-        BUZZ,
-        DUCK,
-        PALINDROMIC,
-        SPY,
-        GAPFUL,
-        SUNNY,
-        SQUARE,
-        JUMPING,
-        HAPPY,
-        SAD*//*;
-
-        private final Checkable check;
-
-        Params(Checkable check) {
-            this.check = check;
+    private static long parseNum(String x) {
+        long num = 0;
+        try {
+            num = Long.parseLong(x);
+        } catch (Exception e) {
         }
+        return num;
     }
-    */
 
     public static void main(String[] args) {
         welcome();
         Scanner scanner = new Scanner(System.in);
-        while (true) { // I suggest splitting this code in small static methods
+        // I suggest splitting this code in small static methods
+        // TODO how to split them in methods
+        // if first method parse input, then I need to return startNumber, count and users, in this case do I need to
+        // create a structure for it
+        // what is the meaning of TODO?
+        // Do I need to create a jira for it?
+        while (true) {
             System.out.println("Enter a request:");
-            String x = scanner.nextLine();
-            String[] result = x.split(" ");
-            if (result.length == 1) {
-                if (x.equals("0")) {
-                    System.out.println("Goodbye!");
-                    break;
-                } else if (Long.parseLong(x) >= 1) {
-                    MyNumber num = new MyNumber(x);
-                    num.printInfo();
-                } else {
-                    System.out.println("The first parameter should be a natural number or zero.");
+            String inputStr = scanner.nextLine();
+
+            if (inputStr.equals("0")) {
+                System.out.println("Goodbye!");
+                break;
+            }
+            String[] input = inputStr.split(" ");
+            long startNumber = parseNum(input[0]);
+            if (startNumber < 1) {
+                System.out.println("The first parameter should be a natural number or zero.");
+                continue;
+            }
+
+            if (input.length == 1) {
+                MyNumber num = new MyNumber(inputStr);
+                num.printInfo();
+                continue;
+            }
+
+            long count = parseNum(input[1]);
+            if (count < 1) {
+                System.out.println("The second parameter should be a natural number.");
+                continue;
+            }
+
+            // TODO how to do it?
+            Collection<String> users = new
```

Comment:
you can do like this:
Collection<String> users = Arrays.stream(Arrays.copyOfRange(input, 2, input.length)).map(String::toUpperCase).collect(Collectors.toSet());

Or as you did:
Collection<String> users = Arrays.asList(input).subList(2, input.length).stream().map(String::toUpperCase).collect(Collectors.toSet());